pkg/client/strategies.go

```diff
@@ -0,0 +1,50 @@
+/*
+Strategy Setup
+
+A utility to abstract the various strategy map ConfigMaps from the service using CRO (RHMI)
+
+Problem Statement:
+ - We require strategy map ConfigMaps to be in place to provide configuration used to provision cloud resources
+ - Each provider overrides null configuration with expected RHMI specific defaults
+ - Non null configuration provided via strategy map is consumed by CRO as configuration for the provisioning of cloud resources
+ - Strategy maps provide the source of truth for the expected state of a cloud resource
+ - Strategy maps are used to update and modify the state of provisioned cloud resources
+
+This utility provides the abstraction necessary, provisioning a strategy map for the infrastructure in which the operator is currently running in
+
+We accept start times for maintenance and backup times as a level of abstraction
+Building the correct maintenance and backup times necessary for specific cloud providers
+*/
+
+package client
+
+import (
+	"context"
+	errorUtil "github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// exported tiers to be used by RHMI operator
+	ProductionTier = "production"
+	DevelopmentTier = "development"
```

Comment:
Small thing, but can we reverse these names, it'll help group them as tiers e.g.

```
TierProduction = ...
TierDevelopment = ... 
```